qan/analyzer/mongo/profiler/monitors.go

```diff
@@ -130,10 +131,10 @@ func (self *monitors) GetAll() map[string]*monitor {
 }
 
 func listDatabases(dialInfo *pmgo.DialInfo, dialer pmgo.Dialer) ([]string, error) {
+	dialInfo.Timeout = MgoTimeoutDialInfo
```

Comment:
This timeout was missing and so `DialWithInfo` was hanging forever when trying to connect to broken replica set.